cpp/tests/test_cubic_spline.cpp

```diff
@@ -0,0 +1,153 @@
+#include <toppra/geometric_path/piecewise_poly_path.hpp>
+#include <toppra/toppra.hpp>
+#include <iostream>
+#include <algorithm>
+#include "gtest/gtest.h"
+
+namespace toppra {
+
+using vvvectors = std::vector<std::vector<value_type>>;
+
+Vectors makevectors(vvvectors v) {
```

Comment:
Currently, this is duplicated from test_hermite_spline.cpp Line 8. I have difficulty in creating the TOPPRA_TEST_LIBRARY that contains this function and for each of these test files (test_cubic_spline.cpp, test_hermite_spline.cpp) to use the function from that library as I am not that familiar with cmake.

Reply:
Isn't it sufficient to replace the function definition in one of the two files by a function declaration ? At linking, the compiler should be able to find the definition in the other file.

Otherwise, you can put this is a separate file that is included by both.

Reply:
I have removed the duplication by moving the function into a separate file. Let me know if you have any suggestions for further improvements. Thanks.